homeassistant/components/togrill/sensor.py

```diff
@@ -0,0 +1,139 @@
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        packet_type=PacketA1Notify.type,
+        packet_extract=_get,
+        entity_supported=_supported,
+    )
+
+
+def _get_version_string(packet: Packet) -> str:
+    assert isinstance(packet, PacketA0Notify)
+    return f"{packet.version_major}.{packet.version_minor}"
+
+
+ENTITY_DESCRIPTIONS = (
+    ToGrillSensorEntityDescription(
+        key="battery",
+        translation_key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        packet_type=PacketA0Notify.type,
+        packet_extract=lambda packet: cast(PacketA0Notify, packet).battery,
+    ),
+    ToGrillSensorEntityDescription(
+        key="version",
+        translation_key="version",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        packet_type=PacketA0Notify.type,
+        packet_extract=_get_version_string,
+    ),
+    _get_temperature_description(1),
+    _get_temperature_description(2),
+    _get_temperature_description(3),
+    _get_temperature_description(4),
+    _get_temperature_description(5),
+    _get_temperature_description(6),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ToGrillConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Gardena Bluetooth sensor based on a config entry."""
+
+    coordinator = entry.runtime_data
+
+    entities: list[ToGrillSensor] = [
+        ToGrillSensor(coordinator, entity_description)
+        for entity_description in ENTITY_DESCRIPTIONS
+        if entity_description.entity_supported(entry.data)
+    ]
+    async_add_entities(entities)
+
+
+class ToGrillSensor(ToGrillEntity, SensorEntity):
+    """Representation of a sensor."""
+
+    entity_description: ToGrillSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: ToGrillCoordinator,
+        entity_description: ToGrillSensorEntityDescription,
+    ) -> None:
+        """Initialize sensor."""
+
+        super().__init__(coordinator, entity_description.packet_type)
+        self.entity_description = entity_description
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = f"{coordinator.address}_{entity_description.key}"
+
+    def _handle_coordinator_update(self) -> None:
+        packet = self.coordinator.data.get(self.entity_description.packet_type)
+        if not packet:
+            self._attr_native_value = None
+            return
+        self._attr_native_value = self.entity_description.packet_extract(packet)
+        self._attr_available = self._attr_native_value is not None
+        super()._handle_coordinator_update()
```

Comment:
Potential bug: The entity always initializes with unknown, even if the coordinator could fetch data, as the attributes are not set after creation. I would recommend to either move to a `property` based approach or to extract the attribute setters to a separate function that you call at the end of the constructor.

(holy shit, now that I am reading it back this sounds like I'm an AI)